Automattic/wp-calypso
# Notices: Redux everything!

Part of ongoing effort to clean up notices:
#888 #1094 #1105 #1313 #1364

This is how notices will look like in the future: #1415

# What is going on here?
Well, we want global notices to be in redux store.

This PR introduces second system for global notices, which relies on redux store.
Because sending notices needs the component to be connected and we have many of those, I want to have both notices systems operational at once so that we can gradually move from old to new one in a couple of PRs.
Then, we will phase out the old one.

This PR provides:
- Store for notices
- Component for displaying them
- Shows notices from `notification-settings` as an example.

# But you're doing it wrong!
I'm open fo suggestions :)

# Plan for now
- Polish and finish this. This will introduce second global notification system and 2 will be active at once. 
- in different PRs gradually switch from old notice system to new one implementing more comple use cases
- Phase out old system

## Why so many changes?
1. `global-notices` have to display notices
2. To send notices, every component sending them needs to be connected
3. To connect those components, `Layout` needs to be wrapped in Redux `Provider`
4. Because `Layout` is now wrapped in Provider, we cannot call `context.layout.setState`
5. We need to move `Layout` state to redux store.

# Testing
All around click-through is much appreciated. There are many new redux providers here and many old components are now reduxified.

The only place the notices work 'the new way' is in:
1. `http://calypso.localhost:3000/me/notifications`
2. Pick a site
3. Change notifications checkbox
4. Save
5. See the Notice 'Settings saved' is there. Awesome!

All the other notices should work also. I hope I didnt break them
